### PR TITLE
Fix engineGetModelTextures memory leak

### PR DESCRIPTION
**Pull request by theSarrum**
Stopping a resource that used engineGetModelTextures does not release the memory used by textures. This pull request solves this problem by setting a parent for the texture.

### CODE REVIEW

File: Client/mods/deathmatch/logic/luadefs/CLuaEngineDefs.cpp [hunk 1]
```diff
@@ -1126,11 +1126,24 @@ int CLuaEngineDefs::EngineGetModelTextures(lua_State* luaVM)
...
         return luaL_error(luaVM, argStream.GetFullErrorMessage());
 
+    CLuaMain*  pLuaMain = m_pLuaManager->GetVirtualMachine(luaVM);
+    CResource* pParentResource = pLuaMain ? pLuaMain->GetResource() : NULL;
+
+    if (!pParentResource)
+    {
+        lua_pushnil(luaVM);
+        return 1;
+    }
```
Comment (qaisjp) [inline Client/mods/deathmatch/logic/luadefs/CLuaEngineDefs.cpp]: Is `pParentResource` ever actually nil?

Comment (ghost) [reply Client/mods/deathmatch/logic/luadefs/CLuaEngineDefs.cpp]: I don't think so. This check is not needed. @theSarrum Please remove this check, we can merge this PR for 1.5.8.

**Review by ghost — COMMENTED** [review]
Good catch, thanks.

**Review by qaisjp — APPROVED** [review]

File: Client/mods/deathmatch/logic/luadefs/CLuaEngineDefs.cpp [hunk 2]
```diff
@@ -1126,11 +1126,18 @@ int CLuaEngineDefs::EngineGetModelTextures(lua_State* luaVM)
     if (argStream.HasErrors())
         return luaL_error(luaVM, argStream.GetFullErrorMessage());
 
+    CLuaMain*  pLuaMain = m_pLuaManager->GetVirtualMachine(luaVM);
+    CResource* pParentResource = pLuaMain ? pLuaMain->GetResource() : NULL;
```
Comment (qaisjp) [inline Client/mods/deathmatch/logic/luadefs/CLuaEngineDefs.cpp]:
```suggestion
    CResource* pParentResource = pLuaMain->GetResource();
```

**Review by qaisjp — COMMENTED** [review]
left a small suggestion but this lgtm

**Review by StrixG — APPROVED** [review]

**Pull request merged**